### PR TITLE
[RNMobile] Update release notes and a comment in embed block

### PR DESCRIPTION
**Pull request by fluiddot**
## Description

This PR updates the release notes, including the missing `1.56.0` section. Additionally, I made an update to a comment from the embed block that was pending as per [this comment](https://github.com/WordPress/gutenberg/pull/31343/files#r662069581).

## How has this been tested?

N/A

## Screenshots 
N/A

## Types of changes

Release notes update

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/embed/edit.js [hunk 1]
```diff
@@ -203,8 +203,8 @@ const EmbedEdit = ( props ) => {
 	};
 
 	const onSubmitNative = ( value ) => {
-		// On native, the URL change is only notified when submitting
-		// so we have to explicitly set the URL.
+		// On native, the URL change is only notified when submitting,
+		// and not via 'onChange', so we have to explicitly set the URL.
```
Comment (fluiddot) [inline packages/block-library/src/embed/edit.js:207]: @hypest I made the comment update that was pending related to [this PR's comment](https://github.com/WordPress/gutenberg/pull/31343/files#r662069581).

File: packages/react-native-editor/CHANGELOG.md [hunk 2]
```diff
@@ -10,13 +10,16 @@ For each user feature we should also add a importance categorization label  to i
...
 
 ## Unreleased
-
 -   [*] Update loading and failed screens for web version of the editor [#32395]
-## 1.55.2
 -   [*] Handle floating keyboard case - Fix issue with the block selector on iPad. [#33089]
--   [**] Fix incorrect block insertion point after blurring the post title field. [#32831]
+-   [*] Embed Block Bottom Sheet to set a link. [#31343]
+
+## 1.56.0
```
Comment (fluiddot) [inline packages/react-native-editor/CHANGELOG.md:16]: I see that there are more changes in the [`1.56.0` GB-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3664) but they're not reflected here 🤔 .

Comment (geriux) [reply packages/react-native-editor/CHANGELOG.md:16]: That's because those were added directly on the GBM PRs and not in the `Gutenberg` ones 😅

Comment (fluiddot) [reply packages/react-native-editor/CHANGELOG.md:16]: Oh ok, thanks for info 🙇 ! Sometimes it's hard to know what landed in each place 😄 .

**Review by geriux — APPROVED** [review]
LGTM! Thanks for updating the changelog!

**Comment by fluiddot** [comment]
Heads up that I removed the "Embed Block Bottom Sheet to set a link" entry from the changelog since this change won't be noticeable by the users because it's behind a dev flag.

**Pull request merged**